templates/cli/src/Client.php.twig

```diff
@@ -0,0 +1,324 @@
+            $this->setPreference(self::PREFERENCE_ENDPOINT, ($endpoint) ? $endpoint : 'http://localhost/v1');
+        }
+
+{% for header in spec.global.headers %}
+{% if header.key != 'Mode' %}
+        if(empty($this->getPreference('{{header.name}}'))) {
+            ${{header.key | caseCamel}} = Console::confirm('Enter your {{header.key | caseLower}} from the Appwrite console: ');
+            if (empty(${{header.key | caseCamel}})) {
+                Console::error("You cannot continue without a {{header.key | caseLower }}. Exiting...");
+                exit();
+            } 
+            $this->setPreference('{{header.name}}', ${{header.key | caseCamel }});
+        }
+
+{% endif %}
+{% endfor %}
+        
+        $result = $this->savePreferences();
+        if ($result === false) {
+            throw new Exception('Could not save preferences.');
+        } else {
+            Console::success('Preferences saved successfully');
+        }
+    }
+
+    /**
+     * Function to load user preferences from
+     * the JSON file
+     */
+    private function loadPreferences(string $filename = self::USER_PREFERENCES_FILE): bool
+    {
+        try {
+            $jsondata = @file_get_contents($filename);
+            if($jsondata === FALSE) {
+                return false;
+            }
+
+            $arr_data = json_decode($jsondata, true);
+            $this->preferences = array_replace($this->preferences, $arr_data);
+            if (!$this->isPreferenceLoaded()) {
+                return false;
+            }
+        } catch (Exception $e) {
+            return false;
+        }
+
+        return true;
+    }
+
+    private function isPreferenceLoaded() : bool {
+        if(empty($this->getPreference(self::PREFERENCE_ENDPOINT))) return false;
+{% for header in spec.global.headers %}
+{% if header.key != 'Mode' %}
+        if(empty($this->getPreference('{{header.name}}'))) return false;
+{% endif %}
+{% endfor %}
+        return true;
+    }
+
+    /**
+     * Function to write user preferences to
+     * the JSON file
+     * 
+     * @return bool
+     */
+    public function savePreferences(string $filename = self::USER_PREFERENCES_FILE): bool
+    {
+        $jsondata = json_encode($this->preferences, JSON_PRETTY_PRINT);
+        $result = file_put_contents($filename, $jsondata);
+        return $result;
+    }
+
+{% for header in spec.global.headers %}
+    /**
+     * Set {{header.key | caseUcfirst}}
+     *
+{% if header.description %}
+     * {{header.description}}
+     *
+{% endif %}
+     * @param string $value
+     *
+     * @return Client
+     */
+    public function set{{header.key | caseUcfirst}}($value)
+    {
+        $this->addHeader('{{header.name}}', $value);
+
+        return $this;
+    }
+
+{% endfor %}
+
+    /***
+     * @param bool $status
+     * @return $this
+     */
+    public function setSelfSigned($status = true)
+    {
+        $this->selfSigned = $status;
```

Comment:
Should we also store this attribute in the prefs file?

Reply:
As discussed I have set the default value to false for the CLI